src/gleam/dict.gleam

```diff
@@ -474,6 +474,65 @@ pub fn upsert(
   |> insert(dict, key, _)
 }
 
+/// Returns a dict where the caller attempts to change a key-value pair in the
+/// dict bei either setting or updating or removing or ignoring a key-value
+/// pair.
+///
+/// In case:
+///
+/// 1. the `update` key is present in the dict, then the value passed to
+///    function `with` is `Some(value)`; then:
+///    - if `fun` returns `Ok(value)`, then the `update` key is newly
+///      associated with to the value of `value` of `Ok(value)`.
+///    - else the `update` key and its associated value are removed from the
+///      dict.
+/// 2. the `update` key is not present in the dict, then the `value` passed to
+///    function `with` is `None`; then:
+///    - if `fun` returns `Ok(value)`, then the `update key` is added to the
+///      dict associated with the `value` of `Ok(value)`.
+///    - else the `update key` is not added to the map.
+///
+/// ## Example
+///
+/// ```gleam
+/// let dict = dict.from_list([#("a", 0), #("b", 1), #("c", 2)])
+///
+/// let inc_if_exists_or_discard = fn(x) {
+///   case x {
+///     Some(i) -> Ok(i + 1)
+///     None -> Error(Nil)
+///   }
+/// }
+///
+/// dict
+/// |> dict.update("a", inc_if_exists_or_discard)
+/// // -> from_list([#("a", 1), #("b", 1), #("c", 2)])
+///
+/// dict
+/// |> dict.update("e", inc_if_exists_or_discard)
+/// // -> from_list([#("a", 0), #("b", 1), #("c", 2)])
```

Comment:
I don't think this shows the functionality very clearly. One example per code block would be good

Reply:
I am not certain I follow. Per which code block?